CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,345 @@
+        toCompressStream << "0x" << std::hex << timeWeights[iGr][i] << " ";
+    }
+    toCompressStream << "\n";
+  }
+
+  toCompressStream << "\n";
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  //  fill  map between xTals and groups. If each xTal is a group there is a one-to-one map
+  const std::vector<DetId>& ebCells = theBarrelGeometry->getValidDetIds(DetId::Ecal, EcalBarrel);
+  std::map<int, int> mapXtalToGroup;
+
+  int iGroup = 0;
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    std::pair<int, int> xTalToGroup(id.rawId(), iGroup);
+    mapXtalToGroup.insert(xTalToGroup);
+    iGroup++;
+  }
+
+  //write to file
+
+  for (std::map<int, int>::const_iterator it = mapXtalToGroup.begin(); it != mapXtalToGroup.end(); it++) {
+    toCompressStream << "CRYSTAL " << dec << it->first << std::endl;
+    toCompressStream << it->second << std::endl;
+  }
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  /////////////////////////////////////
+
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    toCompressStream << "LINCONST " << dec << id.rawId() << std::endl;
+    double theta = theBarrelGeometry->getGeometry(id)->getPosition().theta();
+    EcalLiteDTUPedestalsMap::const_iterator itped = theEcalTPPedestals->getMap().find(id);
+
+    if (itped != theEcalTPPedestals->end()) {
+      peds = &(*itped);
+
+    } else {
+      edm::LogError("EcalEBPhase2TPParamProducer") << " could not find EcalLiteDTUPedestal entry for " << id;
+      throw cms::Exception("could not find pedestals");
+    }
+
+    int shift, mult;
+    double calibCoeff = 1.;
+    bool ok;
+    int tmpPedByGain;
+    for (unsigned int i = 0; i < ecalPh2::NGAINS; ++i) {
+      ok = computeLinearizerParam(theta, gainRatio_[i], calibCoeff, shift, mult);
+      if (!ok) {
+        edm::LogError("EcalEBPhase2TPParamProducer")
+            << "unable to compute the parameters for SM=" << id.ism() << " xt=" << id.ic() << " " << id.rawId();
+        throw cms::Exception("unable to compute the parameters");
+
+      } else {
+        tmpPedByGain = (int)(peds->mean(i) + 0.5);
+        toCompressStream << std::hex << " 0x" << tmpPedByGain << " 0x" << mult << " 0x" << shift << " " << i2cSub_[i]
+                         << std::endl;
+      }
+    }
+  }
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+}
+
+std::vector<int> EcalEBPhase2TPParamProducer::computeWeights(int type) {
+  std::vector<float> sampleSet;
+  std::vector<float> sampleDotSet;
+  std::vector<unsigned int> clockSampleSet;
+  double scaleMatrixBy = 1.;
+
+  switch (nSamplesToUse_) {
```

Comment:
Sorry, there ARE predefined values here ! Both in the python cfi and in the fillDescription ! What else I should be supposed to do ???

Reply:
@perrotta Just a little explanation. Adding a default statement ( so to use our decided default of nSamplesToUse_ =8 and  binOfMaximum_ =6) was not enough because the value of binOfMaximum_ was remaining at the evenbtually wrong value. So we have modified the code so that there is now a local variable lbinOfMaximum initialized to the configured value, which is eventually corrected in a first switch on lbinOfMaximum. The next switch is essentially as it was plus a default fix the default we want. We also added a clear comment in the ecalEBPhase2TPParamProducer_cfi.py about the allowed values of the parameters and what is our defined default. To be committed soon